api/client/usermanager/client_test.go

```diff
@@ -4,170 +4,281 @@
 
-	err = user.Refresh()
+	client := usermanager.NewClientFromCaller(mockFacadeCaller)
+	err := client.DisableUser(user.Name())
 	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(user.IsDisabled(), jc.IsTrue)
 }
 
 func (s *usermanagerSuite) TestDisableUserBadName(c *gc.C) {
-	err := s.usermanager.DisableUser("not!good")
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	client := usermanager.NewClientFromCaller(mockFacadeCaller)
+	err := client.DisableUser("not!good")
 	c.Assert(err, gc.ErrorMatches, `"not!good" is not a valid username`)
 }
 
 func (s *usermanagerSuite) TestEnableUser(c *gc.C) {
-	user := s.Factory.MakeUser(c, &factory.UserParams{Name: "foobar", Disabled: true})
-
-	err := s.usermanager.EnableUser(user.Name())
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	user := names.NewUserTag("foobar")
+	args := params.Entities{Entities: []params.Entity{{user.String()}}}
+	result := new(params.ErrorResults)
+	results := params.ErrorResults{Results: make([]params.ErrorResult, 1)}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("EnableUser", args, result).SetArg(2, results).Return(nil)
+	client := usermanager.NewClientFromCaller(mockFacadeCaller)
+	err := client.EnableUser(user.Name())
 	c.Assert(err, jc.ErrorIsNil)
-
-	err = user.Refresh()
-	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(user.IsDisabled(), jc.IsFalse)
 }
 
 func (s *usermanagerSuite) TestEnableUserBadName(c *gc.C) {
-	err := s.usermanager.EnableUser("not!good")
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	client := usermanager.NewClientFromCaller(mockFacadeCaller)
+	err := client.EnableUser("not!good")
 	c.Assert(err, gc.ErrorMatches, `"not!good" is not a valid username`)
 }
 
 func (s *usermanagerSuite) TestCantRemoveAdminUser(c *gc.C) {
-	err := s.usermanager.DisableUser(s.AdminUserTag(c).Name())
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	admin := names.NewUserTag("admin")
+	args := params.Entities{
+		[]params.Entity{{"user-admin"}},
+	}
+	result := new(params.ErrorResults)
+	results := params.ErrorResults{
+		[]params.ErrorResult{{Error: &params.Error{Message: "failed to disable user: cannot disable controller model owner"}}},
+	}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("DisableUser", args, result).SetArg(2, results).Return(nil)
+
+	client := usermanager.NewClientFromCaller(mockFacadeCaller)
+	err := client.DisableUser(admin.Name())
 	c.Assert(err, gc.ErrorMatches, "failed to disable user: cannot disable controller model owner")
 }
 
 func (s *usermanagerSuite) TestUserInfo(c *gc.C) {
-	user := s.Factory.MakeUser(c, &factory.UserParams{
-		Name: "foobar", DisplayName: "Foo Bar"})
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	now := time.Now()
```

Comment:
there isn't a real need to add time to the imports here as the data is being mocked, and not real. we can skip verifying the result of UserInfo.DateCreated.